src/codegen/codegen_c_visitor.cpp

```diff
@@ -2176,6 +2197,13 @@ void CodegenCVisitor::print_nmodl_constants() {
     }
 }
 
+void CodegenCVisitor::print_mutex_decl() {
+    if (!info.mutex_used) {
+        return;
+    }
+
+    printer->add_line("std::unique_ptr<std::mutex> nmodlmutex;", 2);
```

Comment:
@alkino : FOR PROTCT/MUTEX: I am thinking we should prefer [`omp critical mechanism_name`](https://www.openmp.org/spec-html/5.0/openmpsu89.html) (or omp mutex) here for two reasons 1) just for consistency with openmp used in coreneuron (even though we know it's compatible with std::mutex) 2) all profiling/analysis tools know how to report the performance metrics associated.